src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -260,6 +261,9 @@ class primitive_inst {
 
     std::vector<memory::cptr> _intermediates_memory;
 
+    static const size_t _weights_cache_capacity = 2;
```

Comment:
I think that capacity should be based on global mem size. We need to ensure that (1 + capacity)*total_weights_size < global_mem_size. Otherwise, single weights copy must be stored in the cache. 
`+1` for capacity is needed as we store original constant too

Reply:
For now we can have internal property so we can expose it later if needed